rewrite-javascript/build.gradle.kts

```diff
@@ -83,39 +102,29 @@ tasks.named("check") {
     dependsOn(npmTest)
 }
 
-
 val npmRunBuild = tasks.register<NpmTask>("npmBuild") {
     inputs.files(npmInstall)
         .withPathSensitivity(PathSensitivity.RELATIVE)
     inputs.files(file("rewrite/package.json"))
         .withPathSensitivity(PathSensitivity.RELATIVE)
     inputs.files(fileTree("rewrite/src"))
         .withPathSensitivity(PathSensitivity.RELATIVE)
-    outputs.dir(file("rewrite/dist/src/"))
+    outputs.dir(file("rewrite/dist/"))
+
+    val versionTxt = file("src/main/resources/META-INF/version.txt")
+    outputs.file(versionTxt)
+    doLast {
+        versionTxt.writeText(datedSnapshotVersion)
+    }
```

Comment:
1. Best practice would be to instead create this somewhere in the build directory where it doesn't overlap with another output, then update `processResources` or `jar` to pull the output via the task reference as an input.

If you only want to pull in the version file, then you can specifically call that out via the `include` part of `CopySpec`.

2. The version is presently not an input to this task either. So any time that it changes, but the inputs don't trigger this task, then the file either won't exist or will be incorrect. Should the version writing be it's own task possibly?

Reply:
I want it to work in the IDE as well though.

Reply:
The IDE is able to pull resources from the build folder as well when performing interactions. But I get what you're meaning when you may be using the Intellij Builder as well.